File: src/types/schedule.ts
export type ScheduleStatus = "draft" | "active" | "archived";

export interface Schedule {
  id: number;
  name: string;
  status: ScheduleStatus;
  createdAt: string;
  updatedAt: string;
}

export interface NewSchedule {
  name: string;
  status?: ScheduleStatus;
}

export interface ScheduleEntry {
  id: number;
  scheduleId: number;
  timeSlotId: number;
  classId: number;
  subjectId: number;
  teacherId: number;
  roomId: number;
  decisionLog: string;
  createdAt: string;
}

export interface NewScheduleEntry {
  scheduleId: number;
  timeSlotId: number;
  classId: number;
  subjectId: number;
  teacherId: number;
  roomId: number;
  decisionLog?: string;
}

export interface GenerationResult {
  entriesCreated: number;
  totalScore: number;
  averageScore: number;
  unplacedTasks: UnplacedTask[];
}

export interface UnplacedTask {
  classId: number;
  subjectId: number;
  reason: string;
}

File: src-tauri/src/models/schedule.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Schedule {
    pub id: i64,
    pub name: String,
    pub status: String,
    pub valid_from: Option<String>,
    pub valid_to: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct NewSchedule {
    pub name: String,
    pub status: Option<String>,
    pub valid_from: Option<String>,
    pub valid_to: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ScheduleEntry {
    pub id: i64,
    pub schedule_id: i64,
    pub time_slot_id: i64,
    pub class_id: i64,
    pub subject_id: i64,
    pub teacher_id: i64,
    pub room_id: i64,
    pub decision_log: String,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct NewScheduleEntry {
    pub schedule_id: i64,
    pub time_slot_id: i64,
    pub class_id: i64,
    pub subject_id: i64,
    pub teacher_id: i64,
    pub room_id: i64,
    pub decision_log: Option<String>,
}
